src/dbus_auth_external.erl

```diff
@@ -39,10 +41,64 @@ challenge(_, _) ->
 get_cookie() ->
     case application:get_env(dbus, external_cookie) of
 	undefined -> ?cookie;
+	{ok, system_user} -> 
+			get_current_user_uid();
 	{ok, Val} when is_list(Val) ->
 	    list_to_binary(Val);
 	{ok, Val} when is_binary(Val) ->
 	    Val;
 	{ok, Val} when is_integer(Val) ->
 	    integer_to_binary(Val)
     end.
+
+get_current_user_uid() ->
+	?debug("Getting current USER env~n", []),
+	case os:getenv("USER") of
+			false ->
+					?debug("Missing USER env~n", []),
+					?cookie;
+			User ->
+					% list_to_binary(User)
+					resolve_user_name(User)
+	end.
+
+resolve_user_name(User) ->
+	?debug("Resolving uid for user ~s~n", [User]),
+	Cmd = io_lib:format("id -u ~s", [User]),
+	case run_command(Cmd) of
```

Comment:
why not simply using `os:cmd` ?
I understand you want to check both exit status and stdout, but `id` is unlikely to fail right

Reply:
`os:cmd` doesn't provide a status code, and I found it more difficult to do a parsing of the stdout string than checking for error codes in order to determine success or failure.
`id` ought not to fail, but I'm thinking it might for some set of permissions or conditions especially when it comes to smaller environments (embedded/small images?) - again, I don't have any actual cases to present, it's only that I'm more cautions whenever I have to deal with external commands/executables